src/meta/src/barrier/mod.rs

```diff
@@ -676,14 +686,19 @@ impl GlobalBarrierManager {
                         _ => {}
                     }
                 }
-                resp_result = self.control_stream_manager.next_complete_barrier_response() => {
+                (worker_id, resp_result) = self.control_stream_manager.next_complete_barrier_response() => {
                     match resp_result {
-                        Ok((worker_id, prev_epoch, resp)) => {
-                            self.checkpoint_control.barrier_collected(worker_id, prev_epoch, resp);
+                        Ok(resp) => {
+                            self.checkpoint_control.barrier_collected(resp);
 
                         }
                         Err(e) => {
-                            self.failure_recovery(e).await;
+                            if self.checkpoint_control.on_collect_failure(worker_id, &e)
+                                || self.state.inflight_actor_infos.actor_map.contains_key(&worker_id) {
+                                self.failure_recovery(e).await;
+                            } else {
+                                warn!(e = ?e.as_report(), worker_id, "no barrier to collect from worker, ignore err");
+                            }
```

Comment:
What errors can hit this `warn` branch?

Reply:
When the CN gets disconnected by accidence but there is no actor running on it.